File: cCodes/readability.c
#include <stdio.h>
#include <math.h>

void grade(char frase[]);

int main(void){


    char frase[400];
    printf("Text: ");
    fgets(frase, sizeof(frase), stdin);

    grade(frase);
}

void grade(char frase[]){

    float l;
    float s;

    int i = 0;
    int letters = 0;
    int spaces = 0;
    int sentences = 0;
    while(1){
        if(frase[i] == 0){
            l = (100/((float)spaces+1))*(float)letters; //numero medio de letras a cada 100 palavras
            s = (100/((float)spaces+1))*(float)sentences; //numero médio de sentenças a cada 100 palavras
            break;
        }

        if((frase[i] >= 65 && frase[i] <= 90) || (frase[i] >= 97 && frase[i] <= 122)){
            letters++;
        }
        if(frase[i] == 32){
            spaces++;
        }
        if(frase[i] == 46){ //44 63 e 33
            sentences++;
        }
        i++;
    }

    float index = (0.0588 * l) - (0.296 * s) - 15.8;
    if(index < 1){
        printf("Before Grade 1\n");
    } else if(index > 16){
        printf("Grade 16+\n");
    } else{
        printf("Grade %.0f\n", round(index));
    }
}

File: pythonCodes/readability.py
text = input("Text: ")

letters = 0
words = 1   #number of spaces +1
sentences = 0
for i in text:
    if (65 <= ord(i) <= 90) or (97 <= ord(i) <= 122): #ord converte i para ascii ->>> i recebe o valor de cada caractere antes de ser convertido para ascii
        letters += 1                                                #ex: i = "a" ou i = "J"
    elif ord(i) == 32:
        words += 1
    elif ord(i) in [33, 46, 63]:
        sentences += 1

l = (letters/words)*100 #numero medio de letras a cada 100 palavras
s = (sentences/words)*100 #nuemro medio de sentencas a cada 100 palavras

index = round(0.0588 * l - 0.296 * s - 15.8)
index1 = 0.0588 * l - 0.296 * s - 15.8

if index > 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade: {index}, {index1}")
